ui/src/features/Editor/hooks.ts

```diff
@@ -53,7 +53,22 @@ export default ({
     originPrepend?: string,
   ) => void;
 }) => {
-  const { fitView, screenToFlowPosition } = useReactFlow();
+  const { fitView, screenToFlowPosition, setViewport } = useReactFlow();
+  const { x, y, zoom } = useViewport();
+
+  const latestViewportRef = useRef<{ x: number; y: number; zoom: number }>({
+    x,
+    y,
+    zoom,
+  });
+
+  useEffect(() => {
+    latestViewportRef.current = { x, y, zoom };
+  }, [x, y, zoom]);
```

Comment:
Does this need to be in a useEffect, or can it just be 
```suggestion
  latestViewportRef.current = { x, y, zoom };
```